app/encryption.py

```diff
@@ -28,23 +28,27 @@ class NotificationDictToSign(TypedDict):
 
 
 class CryptoSigner:
-    def init_app(self, app):
-        self.serializer = URLSafeSerializer(app.config.get("SECRET_KEY"))
-        self.salt = app.config.get("DANGEROUS_SALT")
+    def init_app(self, app: Any, secret_key: str | Iterator[str], salt: str) -> None:
+        self.serializer = URLSafeSerializer(secret_key)
+        self.salt = salt
+        self.dangerous_salt = app.config.get("DANGEROUS_SALT")
```

Comment:
So we have two salts to migrate from old one to new one?

Reply:
In a later PR we'll get rid of `DANGEROUS_SALT` entirely (well, move it to `PASSWORD_SALT` that is only used to salt the user passwords before hashing). All `DANGEROUS_SALT` usage will be removed from signing related code.